tests/integration_tests/test_upgrade.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import pytest
+from pathlib import Path
+
+from tests.integration_tests.clouds import ImageSpecification, IntegrationCloud
+from tests.integration_tests.conftest import (
+    get_validated_source,
+    session_start_time,
+)
+
+log = logging.getLogger('integration_testing')
+
+
+def _output_to_compare(instance, file_path, netcfg_path):
+    commands = [
+        'hostname',
+        'dpkg-query --show cloud-init',
+        'cat /run/cloud-init/result.json',
+        'grep Trace /var/log/cloud-init.log',
+        'systemd-analyze',
+        'systemd-analyze blame',
+        'cloud-init analyze show',
+        'cloud-init analyze blame',
+        'cat {}'.format(netcfg_path),
+        'cloud-id'
+    ]
+    with file_path.open('w') as f:
+        for command in commands:
+            f.write('===== {} ====='.format(command) + '\n')
+            f.write(instance.execute(command) + '\n')
+
+
+@pytest.mark.sru_2020_11
+def test_upgrade(session_cloud: IntegrationCloud):
+    source = get_validated_source()
+    if not source.installs_new_version():
+        pytest.skip("Current install method not supported for this test")
```

Comment:
Let's parameterize the {source.value} in this message to tell us specifically what didn't match

Reply:
Skip text isn't actually printed anywhere. There's probably a way to make it show somewhere, but we haven't done it 😂